lib/prima_auth0_ex.ex

```diff
@@ -8,17 +8,17 @@ defmodule PrimaAuth0Ex do
   @doc """
   Obtain a token for the given audience.
   """
-  @spec token_for(String.t()) :: {:ok, String.t()} | {:error, any()}
-  def token_for(audience) do
-    TokenProvider.token_for(TokenProvider, audience)
+  @spec token_for(atom(), String.t()) :: {:ok, String.t()} | {:error, any()}
+  def token_for(client \\ :client, audience) do
```

Comment:
I think it makes sense, especially if we introduce the `config ... :clients` since we won't be able to do `Application.get_env(:prima_auth0_ex, client_name)` anymore